cookbooks/aws-parallelcluster-config/recipes/directory_service.rb

```diff
@@ -49,6 +50,22 @@
     end
   end
 
+  # Password
+  password_secret_arn_str = node['cluster']['directory_service']['password_secret_arn']
+  password_secret_arn = parse_arn(password_secret_arn_str)
+  password_secret_service = password_secret_arn[:service]
+  password_secret_resource = password_secret_arn[:resource]
+  ldap_password =
+    if password_secret_service == "secretsmanager" && password_secret_resource.start_with?("secret")
+      (secret_name = password_secret_resource.split(":")[1])
+      shell_out!("aws secretsmanager get-secret-value --secret-id #{secret_name} --region #{region} --query 'SecretString' --output text").stdout.strip
+    elsif password_secret_service == "ssm" && password_secret_resource.start_with?("parameter")
```

Comment:
This block seems a duplicate of the one in the bash script.

What about defining a bash(python) script to be called by both the places and leave the logic to decide if use secrets manager or ssm in that script?

Reply:
Yes the logic is duplicated in two places, but I do not think it's worth it to put this logic in a common script.
To do that we would put into the AMI a script with this common logic to make it callable by both the recipe and another AMI script. I don't think it's worth it for this logic. Do you know other ways?

Reply:
Nope, I was thinking the same, a file to be used in both the places.